regcomp.c

```diff
@@ -11710,7 +11710,18 @@ S_reg(pTHX_ RExC_state_t *pRExC_state, I32 paren, I32 *flagp, U32 depth)
 	            vFAIL("Expecting close bracket");
 
               gen_recurse_regop:
-                if ( paren == '-' ) {
+                if (paren == '-' || paren == '+') {
+                    if (UNLIKELY(I32_MAX - RExC_npar < num)) {
+                        goto bad_group; /* Avoid overflowing, die
```

Comment:
Rather than add more gotos, I'd be inclined to leave the disposition explicit (and consider making the error message more accurate for this case). If you're worried about duplicating the error string in the binary, I suggest giving it a name.

Reply:
Care to suggest some wording?
Also, what about negative numbers too far from zero.  The comment just above is asking should it be concerned about this from the grok return

Reply:
"Invalid reference to group", perhaps?

I don't see how we can overflow on the negative side: I think at worst we'd substract `-I32_MAX` from 0. It's worth making sure we have test coverage though: I32_MIN or less should give invalid, others should give nonexistent group.

Come to think of it, I'm not sure I understand the handling of `if (num < 1)` at all: `RExC_npar` is our current paren isn't it? So that won't change after a REQUIRE_PARENS_PASS. Unless I have completely the wrong end of the stick, only a '+' can give us a forward reference.

Reply:
I have to think further about your later points in your comment, but a reason I didn't add a new error is because the current code doesn't error on a number that's above I32_MAX, it just sets it to the max value.  It seems to me that using an input number that is out of bounds should give the same error as the one you're proposing.  Agreed?

Reply:
> It seems to me that using an input number that is out of bounds should give the same error as the one you're proposing. Agreed?

Yes, agreed. And though LTA it would not be a disaster if they all gave "nonexistent group"; I'd just prefer to avoid adding gotos to achieve that unless we need to.

Reply:
See updated PR

Reply:
Looks good to me apart from the one new comment.

Reply:
Fixed your point, and pushed